primitives/io/src/lib.rs

```diff
@@ -716,6 +716,22 @@ pub trait Crypto {
 			.map(|sig| ecdsa::Signature::from_slice(sig.as_slice()))
 	}
 
+	/// Sign the given a pre-hashed `msg` with the `ecdsa` key that corresponds to the given public key and
+	/// key type in the keystore.
+	///
+	/// Returns the signature.
+	fn ecdsa_sign_prehashed(
```

Comment:
Might be worth adding `ecdsa_verify_prehashed` for completeness of the API.

AFAICT `recover` already takes a hash, so that one should be fine.

Reply:
Good catch.  Added in latest commit.